feat: handle invalid ephemeral public keys

nventuro opened the pull request:
Closes https://github.com/AztecProtocol/aztec-packages/issues/17158.

This prevents a message with a bad public key x-coord from causing message processing to fail and halt. We still need to handle other such instances of this, including the ciphertext being malformed.

benesjan commented on noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr:
```diff
@@ -337,50 +337,51 @@ impl MessageEncryption for AES128 {
-        // TODO(#17158): handle invalid ephemeral keys when decrypting to prevent DoS vectors
-        let ciphertext_shared_secret = get_shared_secret(recipient, eph_pk.unwrap());
+        // With the sign and the x-coordinate of the ephemeral public key, we can reconstruct the point. This may fail
+        // however, as not all x-coordinates are in the curve. In that case, we simply return `Option::none`.
```
```suggestion
        // however, as not all x-coordinates are on the curve. In that case, we simply return `Option::none`.
```

benesjan commented on noir-projects/aztec-nr/aztec/src/messages/encryption/message_encryption.nr:
```diff
@@ -1,49 +1,48 @@
-    /// the `Contract::process_log` function is designed to work with all the private logs, emitted by a given contract
-    /// and not just by 1 log type.
+    /// Not all ciphertexts are valid - among other things, the ephemeral public key included in it may not correspond
+    /// to a point in the curve. In all such cases, [Option::none] is returned instead.
```
```suggestion
    /// to a point on the curve. In all such cases, [Option::none] is returned instead.
```

benesjan reviewed (approved):
Looks good 👍

nventuro committed:
feat: handle invalid ephemeral public keys

Closes https://github.com/AztecProtocol/aztec-packages/issues/17158.

This prevents a message with a bad public key x-coord from causing message processing to fail and halt. We still need to handle other such instances of this, including the ciphertext being malformed.

Co-authored-by: Nicolás Venturo <nicolas.venturo@gmail.com>
